report_qweb_signer/models/report.py

```diff
@@ -0,0 +1,191 @@
+            'object': obj,
+            'time': time
+        })
+        return filename
+
+    def _attach_signed_read(self, cr, uid, ids, certificate, context=None):
+        if len(ids) != 1:
+            return False
+        filename = self._attach_filename_get(
+            cr, uid, ids, certificate, context=context)
+        if not filename:
+            return False
+        signed = False
+        m_attachment = self.pool['ir.attachment']
+        attach_ids = m_attachment.search(cr, uid, [
+            ('datas_fname', '=', filename),
+            ('res_model', '=', certificate.model_id.model),
+            ('res_id', '=', ids[0])
+        ])
+        if attach_ids:
+            signed = m_attachment.browse(cr, uid, attach_ids[0]).datas
+            signed = base64.decodestring(signed)
+        return signed
+
+    def _attach_signed_write(self, cr, uid, ids, certificate, signed,
+                             context=None):
+        if len(ids) != 1:
+            return False
+        filename = self._attach_filename_get(
+            cr, uid, ids, certificate, context=context)
+        if not filename:
+            return False
+        m_attachment = self.pool['ir.attachment']
+        try:
+            attach_id = m_attachment.create(cr, uid, {
+                'name': filename,
+                'datas': base64.encodestring(signed),
+                'datas_fname': filename,
+                'res_model': certificate.model_id.model,
+                'res_id': ids[0],
+            })
+        except AccessError:
+            raise Warning(
+                _('Saving signed report (PDF):') + '\n',
+                _('You do not have enought access rights to save attachments'))
+        else:
+            _logger.info(
+                "The signed PDF document '%s' is now saved in the database",
+                filename)
+        return attach_id
+
+    def _signer_bin(self, opts):
+        me = os.path.dirname(__file__)
+        return 'java -jar {}/../static/jar/jPdfSign.jar '.format(me) + opts
+
+    def pdf_sign(self, pdf, certificate):
+        pdfsigned = pdf + '.signed.pdf'
+        p12 = _normalize_filepath(certificate.path)
+        passwd = _normalize_filepath(certificate.password_file)
+        if not (p12 and passwd):
+            raise Warning(
+                _('Signing report (PDF)'),
+                _('Certificate or password file not found'))
+        signer_opts = '"%s" "%s" "%s" "%s"' % (p12, pdf, pdfsigned, passwd)
+        signer = self._signer_bin(signer_opts)
+        process = subprocess.Popen(
+            signer, stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True)
+        out, err = process.communicate()
+        if process.returncode:
+            raise Warning(
+                _('Signing report (PDF)'),
+                _('jPdfSign failed (error code: %s). '
+                  'Message: %s') % (str(process.returncode), err))
+        return pdfsigned
+
+    @api.v7
```

Comment:
But ```report``` model declares both:
* API v7: https://github.com/odoo/odoo/blob/9.0/addons/report/models/report.py#L137
* API v8: https://github.com/odoo/odoo/blob/9.0/addons/report/models/report.py#L246

That's why I write in old API, I don't know how to do it in API v8 and allow inheriting in controller: https://github.com/odoo/odoo/blob/9.0/addons/report/controllers/main.py#L45